plutus-playground-server/src/Playground/Interpreter.hs

```diff
@@ -52,6 +52,15 @@ ensureMkFunctionExists script =
             Nothing -> script <> "\n$(mkFunctions [])"
             Just _  -> script
 
+ensureMinimumImports :: (MonadError PlaygroundError m) => SourceCode -> m ()
+ensureMinimumImports script =
+    let scriptString = Text.unpack . Newtype.unpack $ script
+        regex = Regex.mkRegex "^import[ ]+Playground.Contract"
```

Comment:
Should this be `"^import[:blank:]+Playground.Contract" instead, to allow for tabs? 🤔 

And do we need to worry about `qualified`?

Reply:
yes and yes. I'm not sure what to do about qualified because the important thing is that a couple of specific functions are imported. I felt this was the minimum but I could check for either unqualified or importing those functions explicitly. That still leaves the possibility of someone using `as C` and `C.mkFunctions` which is strictly valid but difficult to check for. I don't really want to end up with a massive regex. So I thought a requirement for importing everything would be best.

Reply:
ok, updated so that it will work with `import Playground.Contract` or `import Playground.Contract (...mkFunctions, printSchemas,...)` (order of imports doesn't matter). It won't work with qualified imports but that's tough luck.